sunscreen_tfhe/src/ops/bootstrapping/programmable_bootstrapping.rs

```diff
@@ -340,6 +364,98 @@ pub fn programmable_bootstrap<S>(
     sample_extract(output, cmux_output, 0, glwe_params);
 }
 
+#[allow(clippy::too_many_arguments)]
+/// A generalized version of programmable bootstrapping.
+/// Computes a function `lut` of the encrypted `input`.
+/// However, this generalization features the ability to select which
+/// bits to take during modulus switching. This capability enables
+/// encoding multiple functions into `lut` and bootstrapping each of them
+/// simultaneously.
+///
+/// # Remarks
+/// While [`programmable_bootstrap_univariate`] and
+/// [`programmable_bootstrap_bivariate`] compute a single function of the
+/// input ciphertext, this can compute multiple functions. To do this,
+/// create a [`UnivariateLookupTable`](crate::entities::UnivariateLookupTable) using
+/// [`UnivariateLookupTable::trivivial_multifunctional`](crate::entities::UnivariateLookupTable::trivivial_multifunctional).
+///
+/// `log_v` should equal `ceil(log2(maps.len()))` for the `maps` you
+/// used when creating the LUT.
+///
+/// `log_chi` is the number of most-significant bits to drop during
+/// bootstrapping. Generally, you should set this to zero unless building
+/// other cryptographic primitives, such as Without Padding Bootstrapping
+/// (WoP-PBS)
+pub fn generalized_programmable_bootstrap<S>(
```

Comment:
Why is this not a superset of `programmable_bootstrap_univariate`? It seems like `log_v` would be 1 in the case of 1 map, and then I assume `log_chi` could be set appropriately (I'm guessing 0).